views/pages/RegisterPage.jsx

```diff
@@ -1,280 +1,118 @@
-import React, { Component } from 'react'
+import React, { useState, useEffect, useContext } from 'react'
+import { useForm } from 'react-hook-form'
+import { useNavigate } from 'react-router-dom'
+import { Card, CardContent, CardMedia, Typography } from '@material-ui/core'
+import RegistrationForm from '../forms/RegisterForm'
+import { NotificationContext } from '../contexts'
+import { MIN_WIDTH, VALIDATION_EMAIL_REGEX } from '../../constants'
+import api from '../api'
 
-import Card from '@material-ui/core/Card'
-import CardContent from '@material-ui/core/CardContent'
-import CardMedia from '@material-ui/core/CardMedia'
-import Typography from '@material-ui/core/Typography'
-import TextField from '@material-ui/core/TextField'
-import Button from '@material-ui/core/Button'
-import { withRouter } from '../hoc/withRouter'
-
-import basePathConfig from '../../server/configs/basePathConfig'
-
-const basePath = basePathConfig || ''
-
-class RegisterPage extends Component {
-  constructor(props) {
-    super(props)
-    this.state = {
-      message: '',
+const RegisterPage = ({ classes }) => {
+  const [width, setWidth] = useState(window.innerWidth)
+  const [errors, setErrors] = useState({
+    password: { error: false, message: '' },
+    email: { error: false, message: '' },
+    username: { error: false, message: '' },
+  })
+  const { handleSubmit, control, reset } = useForm({
+    defaultValues: {
       username: '',
       password: '',
+      confirmPassword: '',
       email: '',
-      fullname: '',
-      confirmpw: '',
-      unmatched: false,
-      unInitialized: false,
-      fnInitialized: false,
-    }
-  }
-  componentDidUpdate() {}
-  handleClick = () => {
-    this.setState({
-      open: true,
-    })
-  }
-  handleRequestClose = () => {
-    this.setState({
-      open: false,
-    })
-  }
-  handleChange = (prop) => (event) => {
-    if (prop === 'fullname' && !this.state.fnInitialized) {
-      this.setState({
-        [prop]: event.target.value,
-        fnInitialized: true,
-      })
-      return
-    } else if (prop === 'username' && !this.state.unInitialized) {
-      this.setState({
-        [prop]: event.target.value,
-        unInitialized: true,
-      })
-      return
+      fullName: '',
+    },
+  })
+  const [, setNotification] = useContext(NotificationContext)
+  const navigate = useNavigate()
+  const handleFormSubmit = async (data) => {
+    switch (true) {
```

Comment:
Interesting, I haven't seen this pattern before. So it just executes everything in order? What happens if there's multiple errors?

Reply:
It's waterfall like, but it's easier to read and understand than a bunch of if statements and empty returns. 
Once we update the ui library this should go away as the refs from the react form should work as expected.